pyodide-build/pyodide_build/_f2c_fixes.py

```diff
@@ -5,6 +5,122 @@
 from typing import Iterable, Iterator
 
 
+def prepare_doctest(x):
+    return dedent(x).strip().splitlines(True)
+
+
+def fix_f2c_input(f2c_input_path: str):
+    f2c_input = Path(f2c_input_path)
+    with open(f2c_input) as f:
+        lines = f.readlines()
+    new_lines = []
+    lines = char1_args_to_int(lines)
+
+    for line in lines:
+        line = fix_string_args(line)
+
+        if f2c_input_path.endswith("_flapack-f2pywrappers.f"):
+            line = line.replace("character cmach", "integer cmach")
+            line = line.replace("character norm", "integer norm")
+        if "id_dist" in str(f2c_input):
+            line = line.replace("character*1 jobz", "integer jobz")
+            if "jobz =" in line:
+                line = re.sub("'(.)'", lambda r: str(ord(r.group(1))), line)
+
+        if f2c_input.name in [
+            "_lapack_subroutine_wrappers.f",
+            "_blas_subroutine_wrappers.f",
+        ]:
+            line = line.replace("character", "integer")
+            line = line.replace("ret = chla_transtype(", "call chla_transtype(ret, 1,")
+
+        new_lines.append(line)
+
+    with open(f2c_input_path, "w") as f:
+        f.writelines(new_lines)
+
+
+def fix_string_args(line):
+    line = re.sub("ilaenv", "ilaenvf2pt", line, flags=re.I)
+    if not re.search("call", line, re.I):
+        return line
+    if re.search("xerbla", line, re.I):
+        return re.sub("xerbla", "xerblaf2py", line, flags=re.I)
+    else:
+        return re.sub("'[A-Za-z0-9]'", lambda y: str(ord(y.group(0)[1])), line)
```

Comment:
This substitutes anything that looks like a single character string for the ascii code of the character. The `else:` isn't necessary though since xerbla is always called with multi character strings. `char1_to_int` handles multi character strings. The point is that we want to replace ALL single character strings with their ascii code, and also all multi character strings that appear as arguments to any of the functions that start with c, d, s, or z, and end with one of: 
"gemm", "ggbak", "gghrd", "lacpy", "lamch", "lanhs", "lanst", "larf", "lascl", "laset", "lasr", "ormqr", "orm2r", "steqr", "stevr", "trevc", "trmm", "trsen", "trsv", "unm2r", or "unmqr". 

Basically, these string arguments are used as enums. So to get the machine precision for doubles, you would call `dlamch("E")` where e stands for error or epsilon. As a comment, people add extra letters like `dlamch("Epsilon Machine")` which are ignored by the program.  We want to replace this with `dlamch(69)` where 69 is the ascii code for E. Similarly, see below in  doctests with `ctrmm` and `sTRSV`.

But many multi character strings actually represent strings, so we have a whitelist for when the multi character strings *are* enums. All single character strings that appear as an argument to a function are enums, so this saves me from having to make a longer allow list. Also a few functions have a mix of single character enum args and multi character real string args...